e3db/src/main/java/com/tozny/e3db/Client.java

```diff
@@ -907,6 +910,26 @@ private static LocalEAKInfo toLocalEAK(EAKInfo eakInfo) {
       return new LocalEAKInfo(eakInfo.getKey(), eakInfo.getPublicKey(), eakInfo.getAuthorizerId(), eakInfo.getSignerId(), eakInfo.getSignerSigningKey());
   }
 
+  private Map<String,Object> makeFileMeta(File encryptedFile) throws IOException, NoSuchAlgorithmException {
+    Map<String, Object> fileMeta = new HashMap<>();
+    fileMeta.put("compression", "raw");
+    MessageDigest digest = MessageDigest.getInstance("MD5");
+    FileChannel channel = new FileInputStream(encryptedFile).getChannel();
+    ByteBuffer buffer = ByteBuffer.wrap(new byte[Platform.crypto.getBlockSize()]);
+    try {
+      for (int amt = channel.read(buffer); amt != -1; amt = channel.read(buffer)) {
+        buffer.flip();
+        digest.update(buffer);
+        buffer.clear();
+      }
+    } finally {
+      channel.close();
+    }
+
+    fileMeta.put("checksum", Base64.encodeURL(digest.digest()));
```

Comment:
This should be normal base64 encoding, not base64url encoding. 
https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html
> The base64-encoded 128-bit MD5 digest of the message (without the headers) according to RFC 1864. This header can be used as a message integrity check to verify that the data is the same data that was originally sent. Although it is optional, we recommend using the Content-MD5 mechanism as an end-to-end integrity check. For more information about REST request authentication, see REST Authentication in the Amazon Simple Storage Service Developer Guide. 

RFC link: https://www.ietf.org/rfc/rfc1864.txt

Reply:
This value is sent to E3DB, which AFAIK is Base64-URL encoded. However - the spec doesn't say - https://github.com/tozny/api-specs/blob/master/e3db/e3db.swagger.yml#L1239. @lkwdwrd ?